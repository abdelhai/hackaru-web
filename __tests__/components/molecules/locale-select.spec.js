import { shallowMount } from '@vue/test-utils';
import LocaleSelect from '@/components/molecules/locale-select';

describe('LocaleSelect', () => {
  let factory;
  let wrapper;

  const switchLocalePath = jest.fn();
  const $router = { push: jest.fn() };

  beforeEach(() => {
    factory = () =>
      shallowMount(LocaleSelect, {
        mocks: {
          switchLocalePath,
          $router,
          $i18n: { locale: 'en' }
        }
      });
  });

  describe('when select', () => {
    beforeEach(() => {
      wrapper = factory();
      wrapper.find({ ref: 'base-select' }).vm.$emit('change', 'ja');
    });

    it('change locale', () => {
      expect(switchLocalePath).toHaveBeenCalledWith('ja');
      expect($router.push).toHaveBeenCalled();
    });
  });
});
