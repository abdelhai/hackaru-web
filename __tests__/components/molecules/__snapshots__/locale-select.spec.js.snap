// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleSelect render correctly 1`] = `
<marshmallow-select-stub
  arialabel="Locales"
  value="English"
>
  <option
    value="en"
  >
    
    English
  
  </option>
  <option
    value="ja"
  >
    
    日本語
  
  </option>
</marshmallow-select-stub>
`;
